### PR TITLE
feat(levm): move `levm` EF tests to `cmd/ef_tests` dir

### PR DESCRIPTION
**Pull request by ilitteri**
**Motivation**

We need `ethereum_rust` as a dependency for running the EF tests, specifically for the post-state validation. If we import it inside the crate `levm`, we have a circular dependency as `ethereum_rust` also imports `levm`.

**Description**

- Move `ethereum_rust` EF tests to `cmd/ef_tests/ethereum_rust`.
- Update `ethereum_rust` Makefile targets that run the EF tests according to the previous change
- Move `levm` EF tests to `cmd/ef_tests/levm`.
- Update `ethereum_rust` Makefile targets that run the EF tests to skip `levm` EF tests run (this will be done in the crate `levm`)
- Update `levm` Makefile targets that run the EF tests according to the changes above.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -13,7 +13,8 @@ Cargo.lock
 # MSVC Windows builds of rustc generate these, which store debugging information
 *.pdb
 
-cmd/ef_tests/vectors
+cmd/ef_tests/ethereum_rust/vectors
+cmd/ef_tests/levm/vectors
```
Comment (fmoletta) [inline .gitignore:17]: Question: Aren't the test vectors the same? Can't we use the same files for both tests setups?

File: cmd/ef_tests/levm/ef_tests.rs [hunk 2]
```diff
@@ -0,0 +1,6 @@
+#![allow(clippy::unwrap_used)]
```
Comment (fmoletta) [inline cmd/ef_tests/levm/ef_tests.rs:1]: We should be able to remove this right?

File: .gitignore (hunk 1 above)
Comment (ilitteri) [reply .gitignore:17]: Not quite, we use only a subset of the EF tests for the VM, the `GeneralStateTests` specifically.

Comment (ilitteri) [reply .gitignore:17]: Nevertheless, I think that we can do a proper review of this. I'll file an issue to tackle this.

Comment (ilitteri) [reply .gitignore:17]: Filed in https://github.com/lambdaclass/lambda_ethereum_rust/issues/1173.

Comment (fmoletta) [reply .gitignore:17]: Great! 🚀

**Review by fmoletta — APPROVED** [review]

**Pull request merged**